src/tests/system/tests/test_sssctl.py

```diff
@@ -288,424 +146,293 @@ def test_sssctl__check_ldap_host_object_class_in_domain(client: Client):
+def test_sssctl__auto_private_groups_in_child_domains(client: Client):
     """
-    :title: sssctl config-check detects false positive when auto_private_groups is enabled or disabled in child domains
+    :title: sssctl config-check with auto_private_groups in child domains
     :setup:
-        1. Enable auto_private_groups in child domain
-        2. Disable auto_private_groups in second child domain
-        3. Start SSSD
+        1. Configure auto_private_groups in child domains
     :steps:
-        1. Call sssctl config-check
+        1. Run sssctl config-check
     :expectedresults:
-        1. config-check succeed
+        1. Configuration check succeeds
     :customerscenario: True
     """
-    client.sssd.common.local()
+    setup_local_sssd(client, start=False)
     client.sssd.sssd["domains"] = "td5f4f77.com"
     client.sssd.subdom("td5f4f77.com", "one5f4f77.td5f4f77.com")["auto_private_groups"] = "True"
     client.sssd.subdom("td5f4f77.com", "two5f4f77.td5f4f77.com")["auto_private_groups"] = "False"
-
     client.sssd.start(check_config=False, debug_level=None)
     result = client.sssctl.config_check()
-    assert result.rc == 0, "Config-check failed"
+    assert result.rc == 0, "config-check failed with auto_private_groups"
 
 
-@pytest.mark.tools
-@pytest.mark.ticket(bz=1723273)
-@pytest.mark.topology(KnownTopology.Client)
-def test_sssctl__check_non_default_config_location_missing_snippet_directory(client: Client):
-    """
-    :title: sssctl config-check complains about non existing snippet directory when config is non default
-    :setup:
-        1. Copy sssd.conf file to different directory
-    :steps:
-        1. Call sssctl config-check on that different directory
-        2. Check error message
-    :expectedresults:
-        1. config-check failed
-        2. Error message is properly set
-    :customerscenario: True
-    """
-    client.sssd.common.local()
-    client.sssd.config_apply()
-    client.fs.mkdir("/tmp/test/")
-    client.fs.copy("/etc/sssd/sssd.conf", "/tmp/test/")
-
-    result = client.sssctl.config_check(config="/tmp/test/sssd.conf")
-    assert result.rc != 0, "Config-check successfully finished"
-    assert "Directory /tmp/test/conf.d does not exist" in result.stdout, "Wrong error message on stdout"
+def setup_invalid_option(c):
+    """Configure SSSD with invalid option in non-default config."""
+    setup_local_sssd(c, start=False)
+    c.sssd.default_domain = "local"
+    c.sssd.domain["search_base"] = "True"
+    c.sssd.config_apply(check_config=False)
+    c.fs.mkdir("/tmp/test/")
+    c.fs.copy("/etc/sssd/sssd.conf", "/tmp/test/")
 
 
+@pytest.mark.parametrize(
+    "setup_fn",
+    [
+        # missing snippet dir
+        lambda c: (
+            setup_local_sssd(c, start=False),
+            c.sssd.config_apply(),
+            c.fs.mkdir("/tmp/test/"),
+            c.fs.copy("/etc/sssd/sssd.conf", "/tmp/test/"),
+        ),
```

Comment:
I think we should break this into two tests, missing snippet, invalid permissions.